docs/Coding.md

```diff
@@ -0,0 +1,135 @@
+  currentApiId: string,
+  isExpanded: boolean,
+};
+```
+
+## Intefaces
+
+Interfaces should be used for all public facing API definitions. A table describing when to use interfaces vs. types can be found [here](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/types_or_interfaces).
+
+## Function Components
+
+This project uses function components and hooks over class components. When coding function components in typescript a developer should include any specific props from the **View.props.tsx**
+
+```javascript
+import React, { FunctionComponent } from 'react';
+
+...
+
+export const ExampleComponent: FunctionComponent<ExampleComponentProps> = ({ message, children }: ExampleComponentProps) => (
+  <ReactFragment>
+    <div>{message}</div>
+    <div>{children}</div>
+  </<ReactFragment>>
+);
+```
+
+For components that do not have any additonal props an empty object should be used instead:
+
+```javascript
+import React, { FunctionComponent } from 'react';
+
+...
+
+export const ExampleNoPropsComponent: FunctionComponent<{}> = () => (
+  <div>Example Component with no props</div>
+);
+```
+
+Additonal details around function components can be found [here](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components).
+
+## Hooks
+
+When using hooks with Typescript there are few recommendations that we follow below. Additional recommendations besides the ones mention in this document can be found [here](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks).
+
+### Inference vs Types for useState
+
+Currently we recommend using inference for the primative types booleans, numbers, and string when using useState. Anything other then these 3 types should use a declartive syntax to specify what is expected. For example the following is an example of how to use inference:
+
+```javascript
+const [isEnabled, setIsEnabled] = React.useState(false);
+```
+
+Here is an example how to use a declaritive syntax. When using a declearative syntax if the value can be null that will also need to be specified:
+
+```javascript
+const [user, setUser] = React.useState<IUser | null>(null);
+
+...
+
+setUser(newUser);
+
+
+```
+
+### useReducres
+
+When using reducers make sure you specify the [return type and not use inference](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks#usereducer).
+
+### useEffect
+
+For useEffect only [return the function or undefined](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks#useeffect).
+
+### useRef
+
+When using useRef there are two options with Typescript. The first one is when creating a [read-only ref](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/hooks#useref).
+
+```javascript
+const refExample = useRef<HTMLElement>(null!);
+```
+
+By passing in null! it will prevent Typescript from returning an error saying refExample maybe null.
```

Comment:
Less of a review comment but a typescript questions. How much should the `!` operator be avoided? It seems quite dangerous

Reply:
Wonder if there's an ESLint rule that complains if it's used other than some cases?

Reply:
I would try not to use it except in this case.  I can check to see if there's a better solution.